packages/d2mini/src/indexes.ts

```diff
@@ -124,4 +124,32 @@ export class Index<K, V> {
 
     return new MultiSet(result)
   }
+
+  *lazyJoin<V2>(other: Index<K, V2>): Generator<[[K, [V, V2]], number], void, unknown> {
```

Comment:
Let's remove the duplication between both branches like i did here: https://github.com/electric-sql/d2ts/pull/58/commits/ac24ff08e2bea46dab3fbc00e90b59f3f95d25a1